auth/src/main/java/org/aerogear/mobile/auth/credentials/JwksManager.java

```diff
@@ -104,37 +105,47 @@ public void fetchJwks(@NonNull final KeycloakConfiguration keycloakConfiguration
                     @Nullable final Callback<JsonWebKeySet> callback) {
         String jwksUrl = nonNull(keycloakConfiguration, "keycloakConfiguration").getJwksUrl();
         HttpRequest getRequest = httpModule.newRequest();
-        getRequest.get(jwksUrl);
-        HttpResponse response = getRequest.execute();
-        response.onComplete(() -> {
-            JsonWebKeySet jwks = null;
-            JwksException error = null;
-            // this is invoked on a background thread.
-            if (response.getStatus() == 200) {
-                String jwksContent = response.stringBody();
-                try {
-                    jwks = new JsonWebKeySet(jwksContent);
-                } catch (JoseException e) {
-                    jwks = null;
-                    error = new JwksException(e);
-                    logger.warning("failed to parse JWKS key content. content = " + jwksContent);
+        getRequest.get(jwksUrl).respondWith(new Responder<HttpResponse>() {
+
+
+            @Override
+            public void onResult(HttpResponse response) {
+                JsonWebKeySet jwks = null;
+                JwksException error = null;
+                // this is invoked on a background thread.
+                if (response.getStatus() == 200) {
+                    String jwksContent = response.stringBody();
+                    try {
+                        jwks = new JsonWebKeySet(jwksContent);
+                    } catch (JoseException e) {
+                        jwks = null;
```

Comment:
if an exception is thrown, this is already null

Reply:
Yeah good point.